packages/story-editor/src/components/panels/document/taxonomies/taxonomies.js

```diff
@@ -42,7 +42,9 @@ function TaxonomiesPanel(props) {
 
   const availableTaxonomies = taxonomies.filter((taxonomy) => {
     const isVisible = taxonomy?.visibility?.show_ui;
-    const canAssignTerms = Boolean(capabilities[`assign-${taxonomy?.slug}`]);
+    const canAssignTerms =
+      Boolean(capabilities[`assign-${taxonomy?.slug}`]) ||
+      Boolean(capabilities[`assign-${taxonomy?.restBase}`]);
```

Comment:
I'd check for the `restBase` variant first.

Also you can avoid some repetition by doing

```js
const canAssignTerms =
      Boolean(
        capabilities[`assign-${taxonomy?.slug}`] ||
        capabilities[`assign-${taxonomy?.restBase}`]
      );
```

or even

```js
const canAssignTerms = Boolean(capabilities[`assign-${taxonomy?.restBase || taxonomy?.slug}`])
```

as originally suggested